src/services/download/sentry.rs

```diff
@@ -10,91 +10,86 @@ use std::sync::Arc;
 use std::time::{Duration, Instant};
 
 use actix::{Actor, Addr};
-use actix_web::client::ClientConnector;
+use actix_web::client::{ClientConnector, ClientResponse, SendRequestError};
 use actix_web::{client, HttpMessage};
 use failure::Fail;
-use futures01::future::Either;
-use futures01::prelude::*;
+use futures::compat::{Future01CompatExt, Stream01CompatExt};
+use futures::prelude::*;
 use parking_lot::Mutex;
 use tokio_retry::strategy::{jitter, ExponentialBackoff};
-use tokio_retry::Retry;
 use url::Url;
 
 use super::{DownloadError, DownloadErrorKind, DownloadStatus, USER_AGENT};
 use crate::sources::{FileType, SentryFileId, SentrySourceConfig, SourceFileId};
 use crate::types::ObjectId;
+use crate::utils::futures::{delay, TimedTaskExt};
 
 lazy_static::lazy_static! {
     static ref CLIENT_CONNECTOR: Addr<ClientConnector> = ClientConnector::default().start();
     static ref SENTRY_SEARCH_RESULTS: Mutex<lru::LruCache<SearchQuery, (Instant, Vec<SearchResult>)>> =
         Mutex::new(lru::LruCache::new(100_000));
 }
 
-pub fn download_source(
+async fn start_request(
+    source: &SentrySourceConfig,
+    url: &Url,
+) -> Result<ClientResponse, SendRequestError> {
+    // The timeout is for the entire HTTP download *including* the response stream itself,
+    // in contrast to what the Actix-Web docs say. We have tested this manually.
+    //
+    // The intent is to disable the timeout entirely, but there is no API for that.
+    client::get(url)
+        .with_connector((*CLIENT_CONNECTOR).clone())
```

Comment:
Interesting. We don’t use this in any of the other http-based downloaders. Why is that?

Reply:
Maybe this is only historical to tweak SSL certificates to Sentry or something?

Reply:
This allows us to bypass the restriction for internal IPs that is imposed on the globally registered connector. The rationale is that symbol servers are usually supposed to be external, but Sentry is never. So the default is to restrict all HTTP requests to the internal network, except for the Sentry source. See:
https://github.com/getsentry/symbolicator/blob/56e16ce86170d6d2ba91d530043b7764b7cb9560/src/utils/http.rs#L194

And particularly:
https://github.com/getsentry/symbolicator/blob/56e16ce86170d6d2ba91d530043b7764b7cb9560/src/utils/http.rs#L20

Reply:
Oh, that's important.  I added it to notion page about the client upgrade

Reply:
Thanks, I should've mentioned that much earlier!

Reply:
But where exactly is this hooked up?

https://github.com/getsentry/symbolicator/blob/56e16ce86170d6d2ba91d530043b7764b7cb9560/src/services/download/sentry.rs#L28

This is the only place this is initialized, with the default options? Is this injected somehow via DI? Or overwritten globally?

Reply:
The sentry client uses this exact instance via `with_connector`. All other downloaders use the global connector started via `start_safe_connector`. You can see how this is added to the system registry:

https://github.com/getsentry/symbolicator/blob/56e16ce86170d6d2ba91d530043b7764b7cb9560/src/utils/http.rs#L199

By default, every client request uses the system connector unless explicitly overridden, like above.